turbopuffer/namespace.py

```diff
@@ -30,6 +32,54 @@ def __init__(self, name: str, api_key: Optional[str] = None):
     def __str__(self) -> str:
         return f'tpuf-namespace:{self.name}'
 
+    def __eq__(self, other):
+        if isinstance(other, Namespace):
+            return self.name == other.name and self.backend == other.backend
+        else:
+            return False
+
+    def refresh_metadata(self):
+        response = self.backend.make_api_request('vectors', self.name, method='HEAD')
+        if response.get('status_code') == 200:
+            headers = response.get('headers', dict())
+            dimensions = int(headers.get('x-turbopuffer-dimensions', '0'))
+            approx_count = int(headers.get('x-turbopuffer-approx-num-vectors', '0'))
+            self.metadata = {
+                'exists': dimensions != 0,
+                'dimensions': dimensions,
+                'approx_count': approx_count,
+            }
+        else:
+            self.metadata = {
+                'exists': False,
+                'dimensions': 0,
+                'approx_count': 0,
+            }
+
+    def exists(self) -> bool:
+        """
+        Returns True if the namespace exists, and False if the namespace is missing or empty.
+        """
+        # Always refresh the exists check since metadata from list_namespaces() might be delayed.
+        self.refresh_metadata()
+        return self.metadata['exists']
```

Comment:
![CleanShot 2024-02-07 at 09 36 06@2x](https://github.com/turbopuffer/turbopuffer-python/assets/97400/e4ace472-9fee-441d-912c-78d554762f89)

not sure if you get this too

Reply:
for it to be None, the `refresh_metadata` would have had to raise an error, so this code looks correct to me even if the type checker can't see that

Reply:
What linter are you using that catches that? I haven't seen that.